server/src/services/organization_users.service.ts

```diff
@@ -49,57 +51,41 @@ export class OrganizationUsersService {
     return await this.organizationUsersRepository.update(id, { role });
   }
 
-  async archive(id: string) {
-    await getManager().transaction(async (manager) => {
-      const organizationUser = await manager.findOne(OrganizationUser, {
-        where: { id },
-      });
-      const user = await manager.findOne(User, {
-        where: { id: organizationUser.userId },
-      });
-
-      await this.usersService.throwErrorIfRemovingLastActiveAdmin(user, undefined, organizationUser.organizationId);
-
-      await manager.update(OrganizationUser, id, { status: 'archived', invitationToken: null });
+  async archive(id: string, organizationId: string): Promise<void> {
+    const organizationUser = await this.organizationUsersRepository.findOneOrFail({
+      where: { id, organizationId },
+      relations: ['user'],
     });
 
-    return true;
+    await this.usersService.throwErrorIfRemovingLastActiveAdmin(organizationUser?.user, undefined, organizationId);
+    await this.organizationUsersRepository.update(id, { status: 'archived', invitationToken: null });
   }
 
-  async unarchive(user: User, id: string) {
+  async unarchive(user: User, id: string): Promise<void> {
     const organizationUser = await this.organizationUsersRepository.findOne({
-      where: { id },
+      where: { id, organizationId: user.organizationId },
+      relations: ['user', 'organization'],
     });
-    if (organizationUser.status !== 'archived') return false;
+    if (organizationUser.status !== 'archived') return;
 
     const invitationToken = uuid.v4();
 
-    await getManager().transaction(async (manager) => {
-      await manager.update(OrganizationUser, organizationUser.id, {
-        status: 'invited',
-        invitationToken,
-      });
-      await manager.update(User, organizationUser.userId, { password: uuid.v4() });
-    });
-
-    const updatedUser = await this.usersService.findOne(organizationUser.userId);
+    await this.organizationUsersRepository.update(id, { status: 'invited', invitationToken });
 
-    const currentOrganization: Organization = (
-      await this.organizationUsersRepository.findOne({
-        where: { userId: user.id, organizationId: user.organizationId },
-        relations: ['organization'],
-      })
-    )?.organization;
+    if (this.configService.get<string>('DISABLE_MULTI_WORKSPACE') === 'true') {
+      // Resetting password if single organization
+      await this.usersService.updateUser(id, { password: uuid.v4() });
+    }
 
     await this.emailService.sendOrganizationUserWelcomeEmail(
-      updatedUser.email,
-      updatedUser.firstName,
+      organizationUser?.user?.email,
```

Comment:
We'd always want to expect email, first name etc here.
Wouldn't it make sense to return early incase `organizationUser` is not found rather than using optional chaining?